### PR TITLE
Fixes part of #1132: Add unit test to check TwitterScraper output

### PR DESCRIPTION
**Pull request by vibhcool**
This PR is related to issue #1132

This PR:-
 - Adds Unit Test to compare TwitterScraper.search() method's output with the expected tweet data
 - Changes the access specifiers in TwitterTweet class and MessageEntry class so that they can be tested.
 - Since the output of images during scraping of tweets isn't correct. It is not tested and added in TODO.

### CODE REVIEW

**Comment by mariobehling** [comment]
@vibhcool Please add a link to the related issue into the PR description. Thanks.

**Comment by vibhcool** [comment]
@mariobehling , added issue link in issue description :)

File: test/org/loklak/harvester/TwitterScraperTest.java [hunk 1]
```diff
@@ -83,4 +133,26 @@ public Timeline processTweetList(Timeline[] tweet_list) {
...
+        String k;
+
+        DateTimeFormatter input_format =
+                DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss zzz yyyy");
+        DateTimeFormatter output_format =
+                DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+        LocalDateTime parsed_time = LocalDateTime.parse(time, input_format);
+        ZonedDateTime final_time =
+                ZonedDateTime.of(parsed_time, ZoneId.of("Asia/Kolkata"))
```
Comment (singhpratyush) [inline test/org/loklak/harvester/TwitterScraperTest.java]: Could you please explain this? Why should this hold everywhere?

File: test/org/loklak/harvester/TwitterScraperTest.java [hunk 2]
```diff
@@ -45,34 +60,69 @@ public void testSearch() {
...
         assertEquals(2, ftweet_list.size());
+
+        // Test tweets data with TwitterTweet object
+        TwitterScraper.TwitterTweet tweet = (TwitterScraper.TwitterTweet) ftweet_list.iterator().next();
+
+        assertTrue(String.valueOf(tweet.user).contains(tweet_check.get("user")));
+        assertTrue(String.valueOf(tweet.status_id_url).equals(tweet_check.get("status_id_url")));
+
+        String created_date = dateFormatChange(String.valueOf(tweet.created_at));
+        assertTrue(created_date.equals(tweet_check.get("created_at")));
```
Comment (singhpratyush) [inline test/org/loklak/harvester/TwitterScraperTest.java]: Will this not be affected by timezones?

File: src/org/loklak/harvester/TwitterScraper.java [hunk 3]
```diff
@@ -178,7 +178,7 @@ public static String prepareSearchURL(final String query) {
         String place_id = "", place_name = "";
         boolean parsing_favourite = false, parsing_retweet = false;
         int line = 0; // first line is 1, according to emacs which numbers the first line also as 1
-        boolean debuglog = true;
+        boolean debuglog = false;
```
Comment (vibhcool) [inline src/org/loklak/harvester/TwitterScraper.java]: I shall fix this

File: test/org/loklak/harvester/TwitterScraperTest.java (hunk 1 above)
Comment (vibhcool) [inline test/org/loklak/harvester/TwitterScraperTest.java]: Maybe this is the bug that was bugging the PR, I will get this fixed. :smile:

**Comment by vibhcool** [comment]
Open for review 😄

File: test/org/loklak/harvester/TwitterScraperTest.java (hunk 2 above)
Comment (vibhcool) [inline test/org/loklak/harvester/TwitterScraperTest.java]: fixed this

**Comment by vibhcool** [comment]
@singhpratyush rebased

**Pull request merged**